indico/modules/events/schemas.py

```diff
@@ -5,12 +5,57 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import fields
+import re
+
+from marshmallow import ValidationError, fields, validates_schema
 
 from indico.core.marshmallow import mm
 from indico.modules.events.models.events import Event
 from indico.modules.events.models.principals import EventPrincipal
-from indico.util.marshmallow import PrincipalPermissionList
+from indico.util.i18n import _
+from indico.util.marshmallow import PrincipalPermissionList, not_empty
+
+
+class ValidRegExField(fields.String):
+    """A valid Python regular expression."""
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if len(value) < 2:
+            raise ValidationError(_('Empty regex'))
+        try:
+            return re.compile(value).pattern
+        except ValueError as exc:
+            raise ValidationError(str(exc))
+
+
+class AutoLinkerRuleSchema(mm.Schema):
+    """A regex -> url with placeholders mapping rule."""
+
+    regex = ValidRegExField()
+    url = fields.String(validate=not_empty)
```

Comment:
I would validate this to be a valid URL. Being able to add something like this is nice for testing but probably not particularly useful...

![image](https://github.com/indico/indico/assets/179599/cf1df916-a8c0-4855-8f7c-8b34d3fdd47b)